domain/consensus/reindex_attack_test.go

```diff
@@ -0,0 +1,123 @@
+package consensus_test
+
+import (
+	"fmt"
+	"github.com/kaspanet/kaspad/domain/consensus"
+	"github.com/kaspanet/kaspad/domain/consensus/model/externalapi"
+	"github.com/kaspanet/kaspad/domain/dagconfig"
+	"github.com/kaspanet/kaspad/infrastructure/logger"
+	"math"
+	"math/rand"
+	"os"
+	"testing"
+)
+
+// Test configuration
+const(
+	numBlocksExponent = 14
+	logLevel = "warn"
+	validateMining = false
+)
+
+func LoadJsonDAG(t *testing.T, fileName, testName string, addArbitraryBlocks, useSmallReindexSlack bool) {
+	logger.SetLogLevels(logLevel)
+	params := dagconfig.SimnetParams
```

Comment:
I discussed this with @elichai. It makes no sense here to test all nets because it's a specific DAG loaded from json and we are interested only in it's reachability logic which does not depend on network parameters.